app/lib/features/onboarding/pages/analytics_opt_in_page.dart

```diff
@@ -26,59 +26,237 @@ class AnalyticsOptInPage extends ConsumerWidget {
+          );
+          await telemetryAnalyticsNotifier.setPreference(
+            canReportSentry,
+            newValue,
+            ref,
+          );
+          await telemetryAnalyticsNotifier.setPreference(
+            matomoAnalytics,
+            newValue,
+            ref,
+          );
+          await telemetryAnalyticsNotifier.setPreference(
+            basicTelemetry,
+            newValue,
+            ref,
+          );
+          await telemetryAnalyticsNotifier.setPreference(
+            research,
+            newValue,
+            ref,
+          );
+        }, context),
+        _buildAnalyticsCard(
+          context,
+          lang.sendCrashReportsTitle,
+          lang.sendCrashReportsInfo,
+          allowReportSending,
+          (value) async {
+            await ref
+                .read(analyticsPreferencesProvider.notifier)
+                .setPreference(canReportSentry, value, ref);
+          },
+        ),
+        _buildAnalyticsCard(
+          context,
+          lang.basicTelemetry,
+          lang.basicTelemetryInfo,
+          basicTelemetryEnabled,
+          (value) async {
+            await ref
+                .read(analyticsPreferencesProvider.notifier)
+                .setPreference(basicTelemetry, value, ref);
+          },
+        ),
+        _buildAnalyticsCard(
+          context,
+          lang.appAnalytics,
+          lang.appAnalyticsInfo,
+          matomoAnalyticsEnabled,
+          (value) async {
+            await ref
+                .read(analyticsPreferencesProvider.notifier)
+                .setPreference(matomoAnalytics, value, ref);
+          },
+        ),
+        _buildAnalyticsCard(
+          context,
+          lang.research,
+          lang.researchInfo,
+          researchEnabled,
+          (value) async {
+            await ref
+                .read(analyticsPreferencesProvider.notifier)
+                .setPreference(research, value, ref);
+          },
+        ),
+      ],
+    );
+  }
 
-    return ListTile(
-      onTap: () => toggle(null),
-      title: Text(lang.sendCrashReportsTitle),
-      subtitle: Text(lang.sendCrashReportsInfo),
-      leading: Switch(
-        value: allowReportSending,
-        onChanged: (value) => toggle(value),
+  // Reusable widget for switch toggle option without card
+  Widget _buildToggleAllOption(
+    String text,
+    bool value,
+    Future<void> Function() onToggle,
+    BuildContext context,
+  ) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.end,
+      children: [
+        Text(text),
+        Transform.scale(
+          scale: 0.6,
+          child: Switch(value: value, onChanged: (_) => onToggle()),
+        ),
+      ],
+    );
+  }
+
+  // widget for analytics card with separate preference
+  Widget _buildAnalyticsCard(
+    BuildContext context,
+    String title,
+    String subtitle,
+    bool value,
+    Future<void> Function(bool) onToggle,
+  ) {
```

Comment:
```suggestion
  Widget _buildAnalyticsCard(
  {
    BuildContext context,
    String title,
    String subtitle,
    bool value,
    Future<void> Function(bool) onToggle,
    }
  ) {
```
If there are multiple params in any function then it is better to use named param for better code readability.

Reply:
ok, will change it